### PR TITLE
chore: update to core 42.0.0

### PR DESCRIPTION
**Pull request by bonjourmauko**
Depends on openfisca/openfisca-core#1223

* Minor change.
  - Update OpenFisca-Core to 42.0.0

- - - -

These changes:

- Change non-functional parts of this repository (for instance editing the README)

### CODE REVIEW

**Review by benjello — APPROVED** [review]
Please merge only after all openfisca-france tests are ok.

**Comment by bonjourmauko** [comment]
@benjello I can't merge before the corresponding PR in core is merged.

**Pull request merged**
